### DISCUSSION TITLE
How to use clone/fetch/pull behind a HTTP proxy?

### DISCUSSION THREAD

**Discussion by Guts** [jelmer/dulwich#1182] [Q&A]
I'm working behind an HTTP proxy (set through a `.pac` file). Elsewhere in my project, I'm making HTTP requests using urllib (from std lib) and all my requests work.

Am I missing a parameter to specify to porcelain or should I use a custom client?

**Comment by jelmer** [comment]
By default, Dulwich will pick up proxy settings from the Git configuration but I'm guessing the proxy is just set from the .pac file for you?

dulwich uses urllib3 internally, so any support for .pac in urllib3 would hopefully work out of the box. If not, you could pass in a custom urllib3 pool_manager object that reads the proxy settings from the .pac file.

Does that help?
